vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -88,13 +96,19 @@ type internal ProjectInfoManager
         checkerProvider: FSharpCheckerProvider,
         [<Import(typeof<SVsServiceProvider>)>] serviceProvider: System.IServiceProvider
     ) =
-    // A table of information about projects, excluding single-file projects.  
+
+    // A table of information about projects, excluding single-file projects.
     let projectTable = ConcurrentDictionary<ProjectId, Refreshable<ProjectId[] * FSharpProjectOptions>>()
 
     // A table of information about single-file projects.  Currently we only need the load time of each such file, plus
     // the original options for editing
     let singleFileProjectTable = ConcurrentDictionary<ProjectId, DateTime * FSharpProjectOptions>()
 
+    // Accumulate sorces and references for each project file
+    let lockObject = new Object()
+    let sourceFiles = new Dictionary<string, string[]>()
+    let referenceFiles = new Dictionary<string, string[]>()
```

Comment:
Why have two tables for the information? Just use

    let projectInfoFromCommandLineUpdates = new ConcurrentDictionary<string, string[] * string[]>()

or even 

    let projectInfoFromCommandLineUpdates = 
        new ConcurrentDictionary<string, ImmutableArray<CommandLineSourceFile> * ImmutableArray<CommandLineReference>()

I think I prefer the second - just snapshot the full information we're given since it may be useful in debugging.

Also, in a recent update I renamed ``ProjectInfoManager`` to ``FSharpProjectOptionsManager`` to indicate that it was about maintaining a table mapping projects to up-to-date ``FSharpCompileOptions`` values. It's now been expanded to contain this orthogonal extra information, so perhaps rename back to ``ProjectInfoManager`` or ``ProjectOptionsManager``